File: Dataset/Leetcode/test/110/331.cpp
class Solution {
public:
    bool XXX(TreeNode* root) {
        if(treedepth(root)==-1)return false;
        return true;
    }
    int treedepth(TreeNode* root)//用-1做标志，表示不符合balance-tree
    {
        if(!root)return 0;
        int l=treedepth(root->left),r=treedepth(root->right);
        if(l==-1||r==-1)return -1;
        if(abs(l-r)>1)return -1;
        return max(l,r)+1;
    }
};

File: Dataset/Leetcode/test/112/331.py
 class Solution:
    def XXX(self, root: TreeNode, targetSum: int) -> bool:
        if not root:
            return False
        queue = [root]
        while queue:
            l = len(queue)
            for i in range(l):
                node = queue.pop(0)
                if not node.left and not node.right and node.val == targetSum:
                    return True
                if node.left:
                    node.left.val += node.val
                    queue.append(node.left)
                if node.right:
                    node.right.val += node.val
                    queue.append(node.right)
        return False

